presto-spi/src/main/java/io/prestosql/spi/type/TimeZoneKey.java

```diff
@@ -210,102 +251,89 @@ public static boolean isUtcZoneId(String zoneId)
         return normalizeZoneId(zoneId).equals("utc");
     }
 
-    private static String normalizeZoneId(String originalZoneId)
+    private static String normalizeZoneId(String zoneId)
     {
-        String zoneId = originalZoneId.toLowerCase(ENGLISH);
-
-        boolean startsWithEtc = zoneId.startsWith("etc/");
-        if (startsWithEtc) {
-            zoneId = zoneId.substring(4);
-        }
-
         if (isUtcEquivalentName(zoneId)) {
             return "utc";
         }
 
-        //
-        // Normalize fixed offset time zones.
-        //
+        // The JDK doesn't understand legacy timezones such as Etc/GMT+00:00, GMT-7
+        String lowerCased = zoneId.toLowerCase(ENGLISH);
+        for (String prefix : UTC_OFFSET_PREFIXES) {
+            if (lowerCased.startsWith(prefix)) {
+                int offset = prefix.length();
+                if (lowerCased.charAt(offset) != '+' && lowerCased.charAt(offset) != '-') {
+                    // It must be something like GMT7, or Etc/UTC4, which are invalid
+                    return zoneId;
+                }
+
+                int colon = lowerCased.indexOf(':', offset);
+                int hour;
+                int minute = 0;
+                if (colon != -1) {
+                    hour = Integer.parseInt(lowerCased, offset, colon, 10);
+                    minute = Integer.parseInt(lowerCased, colon + 1, lowerCased.length(), 10);
```

Comment:
What if part before the colon is empty or not digits (`GMT+:10` or `GMT+a:10`)
Same for part after the colon?

Should this all be wrapped in `catch (Exception e) { return zoneId; }`?